hail/src/main/scala/is/hail/io/TextMatrixReader.scala

```diff
@@ -310,27 +313,58 @@ class TextMatrixReader(
       PType.canonical(requestedType.globalType, required = true).asInstanceOf[PStruct]
   }
 
+  def executeGeneric(ctx: ExecuteContext): GenericTableValue = {
+    val fs = ctx.fs
+
+    val tt = fullMatrixType.toTableType(LowerMatrixIR.entriesFieldName, LowerMatrixIR.colsFieldName)
+
+    val lines = GenericLines.read(fs, fileStatuses, params.nPartitions, None, None, params.gzipAsBGZip, false)
```

Comment:
It looks like we're reading twice?  Is that necessary?

Reply:
fixed.